### PR TITLE
Hw 40

### PR DESCRIPTION
**Pull request by SetMyDream**

### CODE REVIEW

**Review by KostiantynVorobiov — APPROVED** [review]
Good

File: src/main/java/com/dev/cinema/service/impl/AuthenticationServiceImpl.java [hunk 1]
```diff
@@ -32,6 +36,8 @@ public User register(String email, String password) {
         User user = new User();
         user.setEmail(email);
         user.setPassword(password);
-        return userService.add(user);
+        userService.add(user);
+        shoppingCartService.registerNewShoppingCart(user);
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/service/impl/AuthenticationServiceImpl.java:40]: user that your are using for registration shopping cart will have id or not?

File: src/main/java/com/dev/cinema/models/ShoppingCart.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+@Entity
+@Table(name = "shopping_cart")
+@NoArgsConstructor
+@Data
+public class ShoppingCart {
+    @Id
+    private Long id;
+    @OneToOne
+    @MapsId
+    @JoinColumn(name = "shopping_cart_id")
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/models/ShoppingCart.java]:
```suggestion
    @JoinColumn(name = "id")
```
its user id too

File: src/main/java/com/dev/cinema/service/impl/AuthenticationServiceImpl.java (hunk 1 above)
Comment (SetMyDream) [reply src/main/java/com/dev/cinema/service/impl/AuthenticationServiceImpl.java:40]: yes, they are

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**